src/base-element.js

```diff
@@ -135,6 +135,18 @@ export class BaseElement {
     return {};
   }
 
+  /**
+   * Subclasses can override this method to indicate that instances need to
+   * use Shadow DOM. The Runtime will ensure that the Shadow DOM polyfill is
+   * installed before uprading and building this class.
+   *
+   * @return {boolean}
+   * @nocollapse
+   */
+  static usesShadowDom() {
+    return false;
```

Comment:
Are you going to override this for relevant classic components in this PR? I believe `amp-next-page` and `amp-truncate-text:0.1` both use shadow dom. Some story components do too but I don't know how relevant that is to this project (don't extend `BaseElement`).

Reply:
Not those for now. Because those components have fallback when SD not available. Do you think maybe I should rename this method to `requiresShadowDom()`?

Reply:
Oh I see. `requiresShadowDom` sounds good to me

Reply:
I went ahead and rename for clarify.